meter/dsmr.go

```diff
@@ -137,10 +139,15 @@ func NewDsmr(uri, energy string, timeout time.Duration) (api.Meter, error) {
 // based on https://github.com/basvdlei/gotsmart/blob/master/gotsmart.go
 func (m *Dsmr) run(conn *bufio.Reader, done chan struct{}) {
 	log := util.NewLogger("dsmr")
+	backoff := backoff.NewExponentialBackOff()
+	backoff.InitialInterval = time.Second
+	backoff.MaxInterval = 5 * time.Minute
 
 	handle := func(op string, err error) {
 		log.ERROR.Printf("%s: %v", op, err)
-		if errors.Is(err, net.ErrClosed) {
+		if err == io.EOF ||
+			errors.Is(err, io.ErrUnexpectedEOF) ||
+			errors.Is(err, net.ErrClosed) {
 			conn = nil
```

Comment:
something crossed my mind. Aren't we leaving a dangling socket by not calling `Close()` here?

Reply:
we actually indeed leave a dangling socket in CLOSED state:
```
$ sudo lsof | grep evcc | grep 3300
evcc      25160               ikruglov    9u     IPv4 0xf8698ffbeed9b957         0t0                 TCP x.x.x.x:59693->x.x.x.x:3300 (CLOSED)
```

however, they are not being accumulated for some reason. So, my short experimentation showed that there no more than one dangling socket after a series of restarts of `ser2net`.

So, it's less a problem than I though originally but I'd suggest to still fix it.

Reply:
@andig, I pushed [31cd0d4](https://github.com/evcc-io/evcc/pull/5641/commits/31cd0d466f546c5911b0059319ccc886037b0f7d) for described problem. Feel free to revert if you think that the problem doesn't worth the amount of changes